hail/python/test/hail/matrixtable/test_matrix_table.py

```diff
@@ -388,6 +388,14 @@ def test_table_join(self):
         self.assertTrue(ds.union_cols(ds.drop(ds.info))
                         .count_rows(), 346)
 
+    def test_table_product_join(self):
+        left = hl.utils.range_matrix_table(5, 1)
+        right = hl.utils.range_table(5)
+        right = right.select(i=hl.range(right.idx + 1, 5)).explode('i').key_by('i')
+        left = left.annotate_rows(matches=right.index(left.row_key, product=True))
+        rows = left.rows()
+        self.assertTrue(rows.all(rows.matches.length() == rows.row_idx))
```

Comment:
Can we add a test case here and in test_table.py that tests the content of the joined array, not just the length? Something like annotating the interval as a value (or adding the interval key to the annotated value in all cases) and checking that